File: 45. Jump Game II/45. Jump Game II.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var jump = function (nums) {
  let ans = 0;
  let left = 0;
  let right = 0;

  while (right < nums.length - 1) {
    let farthest = 0;

    for (let i = left; i < right + 1; i++) {
      farthest = Math.max(farthest, nums[i] + i);
    }

    left = right + 1;
    right = farthest;
    ans++;
  }

  return ans;
};

File: 45. Jump Game II/45. Jump Game II.py
from typing import List


class Solution:
    def jump(self, nums: List[int]) -> int:
        ans = 0
        left = right = 0

        while right < len(nums) - 1:
            farthest = 0

            for i in range(left, right + 1):
                farthest = max(farthest, nums[i] + i)

            left = right + 1
            right = farthest
            ans += 1

        return ans
